playwright/_impl/_async_base.py

```diff
@@ -68,17 +73,19 @@ def _wrap_handler(self, handler: Any) -> Callable[..., None]:
             return mapping.wrap_handler(handler)
         return handler
 
-    def on(self, event: str, f: Any) -> None:
+    def on(self, event: str, f: Callable[..., Union[Awaitable[None], None]]) -> None:
```

Comment:
This will accept both sync and async event listeners

Reply:
Let's only do one thing at a time in each PR. If this PR is about `cast`, it should only touch `cast`.

Reply:
Okay

Reply:
Separated it to https://github.com/microsoft/playwright-python/pull/790